app/presenters/content_item/contents_list.rb

```diff
@@ -13,14 +16,76 @@ def contents_items
       extract_headings_with_ids(body)
     end
 
+    def show_contents_list?
+      return false unless contents_items.count > 1
+      contents_items.count > 2 ||
+        first_item_has_long_table? ||
+        first_item_has_image? ||
+        first_item_has_long_content?
+    end
+
   private
 
     def extract_headings_with_ids(html)
-      headings = Nokogiri::HTML(html).css('h2').map do |heading|
+      headings = parsed_body.css('h2').map do |heading|
         id = heading.attribute('id')
         { text: strip_trailing_colons(heading.text), id: id.value } if id
       end
       headings.compact
     end
+
+    def first_item_has_long_content?
+      first_item_content.count > WORD_COUNT_LIMIT
+    end
+
+    def first_item_content
+      element = first_item.next_element
+      first_item_text = ''
+      until element.name == 'h2'
+        first_item_text += element.text
+        element = element.next_element
+      end
+      first_item_text.split
+    end
+
+    def first_item_has_long_table?
+      table = find_table
+      return unless table.present?
+
+      row_count = table.css('tr').count
+      row_count > 10
+    end
+
+    def find_table
```

Comment:
This looks very similar to the following method, wonder if the logic is similar enough for one method.